rclcpp_lifecycle/src/lifecycle_node.cpp

```diff
@@ -152,8 +152,19 @@ LifecycleNode::LifecycleNode(
 
 LifecycleNode::~LifecycleNode()
 {
+  auto current_state = LifecycleNode::get_current_state().id();
+  if (current_state != lifecycle_msgs::msg::State::PRIMARY_STATE_FINALIZED) {
+    // This might be leaveing sensors and devices without shutting down unintentionally.
```

Comment:
leaving*

Reply:
Good eye! thanks! fixed in https://github.com/ros2/rclcpp/pull/2562/commits/3ad7181de12817cbef965ec22747ec525cefd757